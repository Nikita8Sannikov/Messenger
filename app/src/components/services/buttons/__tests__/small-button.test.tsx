import { render, screen } from "@testing-library/react";

import SmallButtonComponent from "@components/services/buttons/small-button";

describe("SmallButtonComponent", () => {
	test("renders with children", () => {
		render(<SmallButtonComponent>Test Button</SmallButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toBeInTheDocument();
	});

	test("applies small size prop", () => {
		render(<SmallButtonComponent>Test Button</SmallButtonComponent>);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-sizeSmall");
	});

	test("passes through other props correctly", () => {
		render(
			<SmallButtonComponent 
				variant="contained" 
				color="primary"
				disabled
				className="custom-class"
			>
                Test Button
			</SmallButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		expect(button).toHaveClass("MuiButton-contained");
		expect(button).toHaveClass("MuiButton-colorPrimary");
		expect(button).toHaveClass("custom-class");
		expect(button).toBeDisabled();
	});

	test("handles click events", () => {
		const handleClick = jest.fn();
		render(
			<SmallButtonComponent onClick={handleClick}>
                Test Button
			</SmallButtonComponent>,
		);
		const button = screen.getByText("Test Button");
		button.click();
		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	test("matches snapshot", () => {
		const { asFragment } = render(
			<SmallButtonComponent variant="contained" color="primary">
                Test Button
			</SmallButtonComponent>,
		);
		expect(asFragment()).toMatchSnapshot();
	});
});
