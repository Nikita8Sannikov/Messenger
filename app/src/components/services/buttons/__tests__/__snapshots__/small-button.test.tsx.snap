// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SmallButtonComponent matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeSmall MuiButton-containedSizeSmall MuiButton-colorPrimary css-tw6tsg-MuiButtonBase-root-MuiButton-root"
    id=":r4:"
    tabindex="0"
    type="button"
  >
    Test Button
  </button>
</DocumentFragment>
`;
